site/content/docs/02-template-syntax.md

```diff
@@ -887,7 +887,140 @@ Local transitions only play when the block they belong to is created or destroye
 
 ### Animations
 
-TODO i can't remember how any of this works
+```sv
+animate:name
+```
+
+```sv
+animate:name={params}
+```
+
+```js
+animation = (node: HTMLElement, { from: DOMRect, to: DOMRect } , params: any) => {
+	delay?: number,
+	duration?: number,
+	easing?: (t: number) => number,
+	css?: (t: number, u: number) => string,
+	tick?: (t: number, u: number) => void
+}
+```
+
+```js
+DOMRect {
+	bottom: number,
+	height: number,
+	​​left: number,
+	right: number,
+	​top: number,
+	width: number,
+	x: number,
+	y:number
+}
+```
+
+---
+
+An animation is triggered when the contents of a [keyed each block](docs#Each_blocks) are re-ordered. Animations do not run when an element is removed, only when the each block's data is reordered. Animate directives must be on an element that is an *immediate* child of a keyed each block.
+
+Animations will normally be used with the [`flip`](docs#flip) function which will swap the positions of elements when their order changes.
```

Comment:
I don't feel like saying they're normally used with `flip` is very helpful. Or at least I think I'd find it a little misleading. I'd probably prefer this whole thing if it made it clearer that `flip` is just a built-in animation that ships with Svelte.

Seeing as this is reference documentation, there may also be an argument for just introducing the general concept of a custom animation function here, and leaving talking about `flip` for its own section below. This isn't a tutorial, we aren't necessarily trying to ease (editor's note: ba-dum-psh) someone into the concept.

Reply:
This is a good point. My original reasoning for putting it in is that this is the only documentation we have aside from the interactive tutorial. 

Perhaps a mention that the directive can be used with custom animations or Svelte's built-in animations and relevant links rather than any guidance?